### PR TITLE
imgproxy template

### PR DESCRIPTION
**Pull request by ivanonpc-22**
### [URL Generator used for imgproxy](https://progapandist.github.io/imgproxy-form/#)

**Salt** and **Key** can be found in your `Environment` once you create the service. 

[More environment variables](https://docs.imgproxy.net/configuration)

### CODE REVIEW

File: templates/imgproxy/index.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { createTemplate, Services, randomString } from "~templates-utils";
+
+export default createTemplate({
+  name: "imgproxy",
```
Comment (deiucanta) [inline templates/imgproxy/index.ts:4]: The `name` is what is displayed as the label in the panel and on the website. Should this be `Imgproxy`? Maybe this project is an exception and should be lower case.

File: templates/imgproxy/index.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  meta: {
+    description: "imgproxy is the blazing fast and secure image processing tool.",
```
Comment (deiucanta) [inline templates/imgproxy/index.ts:6]: Let's make the description longer. Several ideas from their website

- imgproxy resizes and processes images on the fly and doesn’t consume disk space
- imgproxy uses the world’s fastest image processing library under the hood — libvips. It is screaming fast and has a tiny memory footprint.

File: templates/imgproxy/index.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        type: "string",
+        title: "Domain",
+      },
+    },
+  } as const,
+  generate({ 
+    projectName,
+    serviceName,
+    domain,
+    imgproxykey = randomString(18),
```
Comment (deiucanta) [inline templates/imgproxy/index.ts:43]: Let's use camel case `imgproxyKey`, or simply `key`

File: templates/imgproxy/index.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    imgproxysalt = randomString(18),
```
Comment (deiucanta) [inline templates/imgproxy/index.ts:44]: Same here `imgproxySalt` or simply `salt`

File: templates/imgproxy/index.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    services.push({
+      type: "app",
+      data: {
+        projectName,
+        serviceName: serviceName,
+        env: [
+        `IMGPROXY_KEY=${imgproxykey}`,
+        `IMGPROXY_SALT=${imgproxysalt}`,
+        ].join("\n"),
```
Comment (deiucanta) [inline templates/imgproxy/index.ts:56]: The formatting doesn't seem right. You should install prettier in your code editor and enable "format on save". Let me know if you need futher help with this.

File: templates/index.ts [hunk 6]
```diff
@@ -23,3 +23,4 @@ export { default as uptimekuma } from "./uptimekuma";
 export { default as wikijs } from "./wikijs";
 export { default as wordpress } from "./wordpress";
 export { default as whiteboard } from "./whiteboard";
+export { default as imgproxy } from "./imgproxy";
```
Comment (deiucanta) [inline templates/index.ts:26]: The imports should be ordered alphabetically. I have an option in VS Code that organizes my imports automatically.

File: templates/imgproxy/index.ts (hunk 1 above)
Comment (deiucanta) [reply templates/imgproxy/index.ts:4]: Let's leave it lowercase for this, but keep in mind for others. In the "whiteboard" template I made it `Whiteboard`.

**Review by deiucanta — CHANGES_REQUESTED** [review]

**Comment by deiucanta** [comment]
> ### [imgproxy.net](https://imgproxy.net/)
> ### [URL Generator used for imgproxy](https://progapandist.github.io/imgproxy-form/#)
> **Salt** and **Key** can be found in your `Environment` once you create the service.
> 
> [More environment variables](https://docs.imgproxy.net/configuration)

Nice instructions. There is a property called `instructions` in the `meta` field. Those will appear after the template is used to create an app. That is good for telling the user how to get started, what are the default credentials, where to find different things (env vars, etc).

**Comment by ivanonpc-22** [comment]
Sounds good I will get working on these changes.

File: templates/imgproxy/index.ts (hunk 1 above)
Comment (ivanonpc-22) [reply templates/imgproxy/index.ts:4]: Sounds good.

**Pull request closed without merging**